tests/components/number/test_init.py

```diff
@@ -46,3 +67,49 @@ async def test_sync_set_value(hass):
 
     assert number.set_value.called
     assert number.set_value.call_args[0][0] == 42
+
+
+async def test_custom_integration_and_validation(
+    hass, entity_reg, enable_custom_integrations
+):
+    """Test we can only set valid values."""
+    platform = getattr(hass.components, f"test.{DOMAIN}")
+    platform.init()
+
+    reg_entry_1 = entity_reg.async_get_or_create(
+        DOMAIN,
+        "test",
+        UNIQUE_NUMBER,
+    )
+
+    assert await async_setup_component(hass, DOMAIN, {DOMAIN: {CONF_PLATFORM: "test"}})
+    await hass.async_block_till_done()
+
+    state = hass.states.get(reg_entry_1.entity_id)
+    assert state.attributes.get(ATTR_VALUE) is None
```

Comment:
Assert the main state instead. It should be 50.0.
```suggestion
    assert state.state == "50.0"
```